src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -1,255 +1,251 @@
+                cartEntry.setCart(cart);
             }
 
-            cart.setUser(user);
+        for (int i = 0; i < quantity; i++) {
+            productTotal -= productDTO.getPrice();
+        }
 
-            cart.setProductList(productList);
-            cart.setLastUpdatedOn(LocalDateTime.now());
-            cart.setTotal(cart.getTotal() - productTotal);
+        cart.setEntries(cartEntries);
+        cart.setTotal(cart.getTotal() + productTotal);
 
-            cartRepository.save(cart);
+        cartRepository.save(cart);
 
-            return CartMapper.cartEntityToDto(cart);
-        }
+        return cartEntityToDto(cartRepository.findByUserId(user.getId()));
     }
 
     public void deleteCartByEmail(String email) throws  EntityNotFoundException {
-
-        User user;
+        UserDTO user;
         try {
-            user = UserMapper.DtoToUserEntity(userServiceClient.callGetUserByEmail(email));
+            user = userServiceClient.callGetUserByEmail(email);
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
-
-        //TODO you might benefit from hibernate caching but you might also do the same DB query twice here(once when you want to see it's not null and 2nd time when you use it)
-        if (cartRepository.findByUser(user) != null) {
-            Cart cart = cartRepository.findByUser(user);
-
-            List<Product> prodList = cart.getProductList();
-
-            Iterator<Product> iter = prodList.iterator();
-
-            while (iter.hasNext()) {
-                Product product = iter.next();
-                iter.remove();
-
-                log.info("{} succesfully deleted from Product List",product.getProductName());
-
-            }
+        Cart cart = cartRepository.findByUserId(user.getId());
             cartRepository.delete(cart);
-            log.info("Cart succesfully deleted!");
-        }
-        log.info("Cart is empty (null)!");
+            log.info("Cart successfully deleted!");
     }
 
     public CartDTO getCartByEmail(String email) throws EntityNotFoundException {
 
-        User user;
+        UserDTO user;
         try {
-            user = UserMapper.DtoToUserEntity(userServiceClient.callGetUserByEmail(email));
+              user = userServiceClient.callGetUserByEmail(email);
 
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
 
-        if (cartRepository.findByUser(user) != null) {
-            Cart cart = cartRepository.findByUser(user);
+        if (cartRepository.findByUserId(user.getId()) != null) {
+            Cart cart = cartRepository.findByUserId(user.getId());
 
-            return CartMapper.cartEntityToDto(cart);
+            return cartEntityToDto(cart);
         } else
             return null;
```

Comment:
it's not very nice to return null. better throw a not found exception

Reply:
Done!